src/hooks/core/chat/useSystemMessage.ts

```diff
@@ -57,38 +60,52 @@ const useSystemMessage = () => {
     saveSystemMessageCallback: SaveSystemMessageCallback
   ): Promise<boolean> {
     // If the callback is not provided, do nothing
+    featLog('checkpoint1');
     if (!saveSystemMessageCallback) return false;
 
+    featLog('checkpoint2');
     // If the message is a follow topic message, do nothing
     if (message?.text?.toLocaleLowerCase()?.includes('this topic has new')) return false;
 
+    featLog('checkpoint3');
     // If the message is a system message, save the message
     if (__isMessageOnlyForSystemUser(message) && __isMySystemMessage(message)) {
       const newMessage = checkSystemMessageOnlyForSystemUser(message);
       saveSystemMessageCallback?.(newMessage);
       return true;
     }
 
+    featLog('checkpoint 3.5');
     if (__isMessageOnlyForSystemUser(message) && !__isMySystemMessage(message)) return false;
 
+    featLog('checkpoint4');
     if (__isMessageForOtherUser(message)) {
       const newMessage = checkSystemMessageOnlyForOtherSystemUser(message);
       saveSystemMessageCallback?.(newMessage);
       return true;
     }
 
+    featLog('checkpoint5');
     // If the message is a system message, save the message
     if (__isSystemMessage(message)) {
       const newMessage = checkSystemMessage(message);
       saveSystemMessageCallback?.(newMessage);
       return true;
     }
 
+    featLog('checkpoint6');
     if (message?.better_type) {
       saveSystemMessageCallback?.(message);
       return true;
     }
 
+    featLog('checkpoint7');
+    if (message?.type === 'regular') {
+      saveSystemMessageCallback?.(message);
+      return true;
+    }
+
+    featLog('checkpoint8');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of logging statements at various checkpoints in the function is a good practice for debugging and understanding the flow of execution. However, it's important to ensure that these logs are not too verbose or contain sensitive information when running in a production environment. Consider adding an environment check to only log these messages in a development environment.
```diff
+    if (process.env.NODE_ENV === 'development') {
        featLog('checkpoint1');
+    }
```
Remember to add this check for all the `featLog` calls in this hunk.


<!-- This is an auto-generated comment by OSS CodeRabbit -->